config/backend-config/backend-config.go

```diff
@@ -323,19 +323,27 @@ func (bc *backendConfigImpl) StartWithIDs(ctx context.Context, workspaces string
 	bc.ctx = ctx
 	bc.cancel = cancel
 	bc.blockChan = make(chan struct{})
-	bc.cache = cacheOverride
-	if bc.cache == nil {
-		identifier := bc.Identity()
-		u, _ := identifier.BasicAuth()
-		secret := sha256.Sum256([]byte(u))
-		cacheKey := identifier.ID()
-		bc.cache = cache.Start(
-			ctx,
-			secret,
-			cacheKey,
-			bc.Subscribe(ctx, TopicBackendConfig),
-		)
+
+	// setup cache
+	identifier := bc.Identity()
+	u, _ := identifier.BasicAuth()
+	secret := sha256.Sum256([]byte(u))
+	cacheKey := identifier.ID()
+	dbCache := cache.Start(
+		ctx,
+		secret,
+		cacheKey,
+		bc.Subscribe(ctx, TopicBackendConfig),
+	)
+
+	if dbCache != nil {
+		bc.cache = dbCache
+	} else {
+		pkgLogger.Info(`couldn't setup config cache in database, disabling cache`)
+		disableCache()
+		bc.cache = cacheOverride
 	}
```

Comment:
To avoid running unnecessary code if `bc.cache` is set:

```suggestion
	// setup cache
	if bc.cache == nil {
		identifier := bc.Identity()
		u, _ := identifier.BasicAuth()
		secret := sha256.Sum256([]byte(u))
		cacheKey := identifier.ID()
		bc.cache = cache.Start(
			ctx,
			secret,
			cacheKey,
			bc.Subscribe(ctx, TopicBackendConfig),
		)
		if bc.cache == nil {
			bc.cache = cacheOverride
			pkgLogger.Info("Couldn't setup config cache in database, disabling cache")
			disableCache()
		}
	}
```